pkg/task/task.go

```diff
@@ -147,3 +222,41 @@ type CheckpointInfo struct {
 type CompletedTask struct {
 	Checkpoint TaskCheckpoint `json:"checkpoint"`
 }
+
+func newCheckpointInfo() CheckpointInfo {
+	return CheckpointInfo{
+		CompletedTasks: make(map[string]*CompletedTask),
+	}
+}
+
+func (c CheckpointInfo) taskCompleted(name string, completedTask *CompletedTask) {
+	c.CompletedTasks[name] = completedTask
+}
+
+func getCheckpointFile(file string) (*CheckpointInfo, error) {
```

Comment:
maybe something like:
```suggestion
func readCheckpointFile(file string) (*CheckpointInfo, error) {
```